### ISSUE TITLE
UI should be deployable and run independently from a cluster

### ISSUE THREAD

**Issue by eriknelson** [fusor/mig-ui#258]
A user should be able to deploy the UI easily into a cluster using some kind of installation script, whether that's ansible or bash. We need a way to provide configuration data to the front end via `migMeta` (things like the clusterAPI address, where to create secrets, what namespace the API objects should be created in, etc.). Webpack delivers this data to the front-end by reading a local `config.dev.json` file currently, but inside of a cluster, this is going to require a lightweight server that has a few responsibilities:

* Serve static, like the production js bundle, fonts, images, etc.
* When the UI is requested, dynamically read configuration data from something like a `ConfigMap`, encode that data and anything else that the UI would need to know into a base64 encoded `migMeta` object, and then serve the apps html template with that encoded blob + the script tag to load the app.

Overall, this issue encompasses:
* Writing the mig-ui-server
* Ensuring the production build is working (Closes #154)
* Building a mig-ui image that packages everything we need and runs the mig-ui-server
* Working through any integration issues that crop up

**Comment by eriknelson** [comment]
I had a branch that I had done a bunch of work on the serverside of this here: https://github.com/eriknelson/mig-ui/commit/48f3c6434bcdca62e59d0ac62a0bafeac353c608

Plan is to rebase and use what's here.

**Comment by eriknelson** [comment]
WIP branch here: https://github.com/fusor/mig-ui/pull/265

Server + Image + Ansible and templates are mostly complete. We are currently blocked on testing, because CORS is still not supported on 4.x clusters. We're hopeful this will be implemented for us relatively quickly so we will be able to test this.

**Comment by eriknelson** [comment]
An unsupported solution is in place and confirmed to be working, we have a JIRA against OCP requesting CORS support be officially supported in the 4.2 timeframe. I've also confirmed the 3.x CORS configuration works as expected, so I was able to perform a full login and migration from a deployed UI in a 4.1 cluster.

We're still seeing some static file issues with the CSS that's baked into the production build. Working to resolve that as the last integration problem before merge to master.
